### PR TITLE
Add 'get_vowel', improve profile and help

### PR DESCRIPTION
**Pull request by maxsaltonstall**
Implement vowel-only letter option
More helpful new player guidance
Level-error prevention on profile command

### CODE REVIEW

**Comment by maxsaltonstall** [comment]
I'm a terrible person. I did three separate things in one PR

File: bot/models/player.py [hunk 1]
```diff
@@ -127,11 +138,18 @@ async def show_progress(self, ctx):
 
         letter_xp_string = StringUtil.format_player_xp(self.get_letter_xp())
 
+        try:
```
Comment (davidstanke) [inline bot/models/player.py:141]: Can we move this logic into the `get_level` method?

File: bot/models/player.py [hunk 2]
```diff
@@ -127,11 +138,18 @@ async def show_progress(self, ctx):
...
+            logging.error("# Error 5 #: Error when getting player level")
+            logging.exception(str(e))
+            lvl = 1
+
         embed = discord.Embed(
             title=f"Player profile: {self.get_username()}",
             description=f"""
                 ...................................................
-                Level: **{self.get_level()}**
+                Level: **{lvl}**
```
Comment (davidstanke) [inline bot/models/player.py:152]: ...so that this can instead be `self.get_level()` ...IMHO it's cleaner that way.

File: bot/models/player.py [hunk 3]
```diff
@@ -81,6 +81,17 @@ def add_letter(self):
                 return(f"{self.state['username']}, you can have an **{letter}**")
             return(f"{self.state['username']}, you can have a **{letter}**")
 
+    def add_vowel(self):
```
Comment (davidstanke) [inline bot/models/player.py:84]: I'd prefer to see this refactored into a variant of the `add_letter` method. So we'd have a method:
`add_letter(self, restricted_letters: list=[])` ... which by default would have no restricted letters.

Then `add_vowel` would call `add_letter`, and specify the vowels as restricted letters.

Also note that (IMHO?) `add_vowel` shouldn't create duplicates, so it should also respect the letters in hand. We can do that by appending the consonants to the letters in hand when requesting a letter.

Another approach -- which might be better? Would be to have a parameter on `add_letter`: e.g. "type" -- which would default to any, but could be restricted to vowel (or other).

Obvs I have opinions here! LMK if you'd like to chat about it, or alternatively, we push forward as-is and you can assign me an issue to do this refactor.

**Review by davidstanke — CHANGES_REQUESTED** [review]

File: bot/models/player.py (hunk 1 above)
Comment (maxsaltonstall) [reply bot/models/player.py:141]: Sure thing, good call.

Comment (maxsaltonstall) [reply bot/models/player.py:141]: Done

File: bot/models/player.py (hunk 2 above)
Comment (maxsaltonstall) [reply bot/models/player.py:152]: Done

File: bot/models/player.py (hunk 3 above)
Comment (maxsaltonstall) [reply bot/models/player.py:84]: I'm fine with any of those changes, and agree on duplicate prevention for add_vowel. Mind if I punt that follow-up CL to you?

Comment (maxsaltonstall) [reply bot/models/player.py:84]: New issue created

**Review by davidstanke — APPROVED** [review]

File: bot/models/player.py (hunk 3 above)
Comment (davidstanke) [reply bot/models/player.py:84]: Cool thanks -- on it.

**Comment by davidstanke** [comment]
Merging this so I can work on #169

**Pull request merged**